x-pack/plugins/cases/public/components/all_cases/columns.tsx

```diff
@@ -51,35 +53,106 @@ const TagWrapper = styled(EuiBadgeGroup)`
 const renderStringField = (field: string, dataTestSubj: string) =>
   field != null ? <span data-test-subj={dataTestSubj}>{field}</span> : getEmptyTagValue();
 
-interface GetCasesColumn {
-  actions: Array<DefaultItemIconButtonAction<Case>>;
-  caseDetailsNavigation: CasesNavigation<CaseDetailsHrefSchema, 'configurable'>;
+export interface GetCasesColumn {
+  caseDetailsNavigation?: CasesNavigation<CaseDetailsHrefSchema, 'configurable'>;
   filterStatus: string;
-  isModal: boolean;
+  handleIsLoading: (a: boolean) => void;
+  refreshCases?: (a?: boolean) => void;
+  showActions: boolean;
 }
-export const getCasesColumns = ({
-  actions,
+export const useCasesColumns = ({
   caseDetailsNavigation,
   filterStatus,
-  isModal,
+  handleIsLoading,
+  refreshCases,
+  showActions,
 }: GetCasesColumn): CasesColumns[] => {
-  const columns = [
+  const { loading: isLoadingCases, dispatchUpdateCaseProperty } = useGetCases();
+  // Delete case
+  const {
+    dispatchResetIsDeleted,
+    handleOnDeleteConfirm,
+    handleToggleModal,
+    isDeleted,
+    isDisplayConfirmDeleteModal,
+    isLoading: isDeleting,
+  } = useDeleteCases();
+
+  const [deleteThisCase, setDeleteThisCase] = useState<DeleteCase>({
+    id: '',
+    title: '',
+    type: null,
+  });
+  const confirmDeleteModal = useMemo(
```

Comment:
We can just put this component directly in the `return()` and if update `ConfirmDeleteModal` to `ConfirmDeleteModal = React.memo(() => {})`. But I can handle that in my cleanup PR if you don't want to do it here